src/test/java/com/microsoft/sqlserver/jdbc/MaxResultBufferTest.java

```diff
@@ -0,0 +1,280 @@
+            try (ResultSet resultSet = statement.getResultSet()) {
+                while (resultSet.next()) {}
+            }
+        }
+    }
+
+    /**
+     * This method tests if Statements are detached properly, when first one hasn't been completely retrieved and second
+     * one have been executed
+     *
+     * @throws SQLException
+     *         Exception is thrown when maxResultBuffer is exceeded
+     */
+    private void testPreparedStatementWithMultipleResultSets() throws SQLException {
+        String selectSQL = "SELECT * FROM " + TEST_TABLE_NAME;
+
+        try (Connection connection = DriverManager.getConnection(connectionString);
+                PreparedStatement statement = connection.prepareStatement(selectSQL);
+                ResultSet resultSet = statement.executeQuery()) {
+
+            try (PreparedStatement secondStatement = connection.prepareStatement(selectSQL);
+                    ResultSet secondResultSet = secondStatement.executeQuery()) {
+                while (resultSet.next()) {}
+
+                try (PreparedStatement thirdStatement = connection.prepareStatement(selectSQL);
+                        ResultSet thirdResultSet = thirdStatement.executeQuery()) {
+                    while (thirdResultSet.next()) {}
+                    while (secondResultSet.next()) {}
+                }
+            }
+        }
+    }
+
+    /**
+     * This method tests if ResultSet's are retrieved correctly, when more than one Query is executed inside single
+     * statement
+     * 
+     * @throws SQLException
+     *         Exception is thrown when maxResultBuffer is exceeded
+     */
+    private void testTwoQueriesInOneStatement() throws SQLException {
+        try (Connection connection = DriverManager.getConnection(connectionString);
+                Statement statement = connection.createStatement()) {
+            statement.execute("SELECT * FROM " + TEST_TABLE_NAME + ";SELECT * FROM " + TEST_TABLE_NAME);
+
+            try (ResultSet resultSet = statement.getResultSet()) {
+                while (resultSet.next()) {}
+            }
+
+            if (statement.getMoreResults()) {
+                try (ResultSet totallyNewResultSet = statement.getResultSet()) {
+                    while (totallyNewResultSet.next()) {}
+                }
+            }
+        }
+    }
+
+    private static String generateRandomString(int precision) {
+        int leftLimit = 33;
+        int rightLimit = 126;
+        Random random = new Random();
+        return random.ints(leftLimit, rightLimit).limit(precision)
+                .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append).toString();
+    }
+
+    private static void setResponseBufferingAdaptive(boolean adaptive) {
+        String value = adaptive ? "adaptive" : "full";
+        connectionString = TestUtils.addOrOverrideProperty(connectionString, "responseBuffering", value);
```

Comment:
Because connectionString is defined in AbstractTest and is static, this line will change the connection string for all the tests. Please use a local connection string .